edx/cc2olx
# [BD-27] [BB-2680] Polish and prepare cc2olx converter

This PR lays ground for rework of https://github.com/openedx/cc2olx tool.

1. Added `setup.py` configuration, so package now can be installed and used like `cc2olx -i <FILE>`.
1. Added `tox` configuration, so all tests and linting can be run with `tox`.
1. Removed usage of raw `os.path` favoring `pathlib`.
1. Slightly updated readme.
1. Added `Makefile` with some utility commands.
1. Simplified `argparse` interface to have only `--inputs` argument, for both directories and files.
1. Reformatted code and removed some unused lines.
1. Removed `Cartridge.serialize` method. I'm not sure about it, but it seems that methods from `olx` module do the same.
1. Implemented zip output mode.
1. Covered mostly everything with tests.


**JIRA tickets**:
- [BLENDED-406](https://openedx.atlassian.net/browse/BLENDED-406)


**Dependencies**: None

**Testing instructions**:

1. Check that all tests and linting are passing:
    ```sh
    git clone https://github.com/edx/cc2olx
    cd cc2olx
    make test-all
    ```

2. Check that tool converts some of the test `.imscc` files without errors: 
    ```sh
    virtualenv -p python3 .fenv
    . .fenv/bin/activate
    python setup.py install
    cc2olx -i test_data/single-page.imscc
    ```

    After this, you should see new `output` directory in your current directory. It should contain `single-page.tar.gz` file.

3. Open any course in studio, then click on `Tools` -> `Import` and try to import this `single-page.tar.gz`. You should see, that your course is replaced with new imported course.


**Author notes and concerns**:

1. As @kaizoku mentioned, I'm not sure if sample `.imscc` files can be distributed.
1. I'm not sure what's the purpose of `zip` output mode, so I archive whole workspace directory.

**Reviewers**
